### ISSUE TITLE
pyorient not working with orientdb 3.0.0

### ISSUE THREAD

**Issue by rrmerugu** [mogui/pyorient#270]
When i try to establish a connection to the db from pyorient, the below error is showing up. im using pyorient==1.5.5, is it possible its not compatible with orientdb 3.0.0. Any plans to support pyorient for orientdb 3.0.0 version soon ?

> pyorient.exceptions.PyOrientWrongProtocolVersionException: Protocol version 37 is not supported yet by this client.

PS: This is the version of the orientdb i'm using https://orientdb.com/download.php?file=orientdb-community-gremlin-3.0.0.tar.gz

**Comment by SubhashKaviya** [comment]
can you share your code please?

**Comment by XhrCkYsSgZokJL** [comment]
pyorient.OrientDB('localhost', 2424) 
session_id = self.client.connect('user', 'password')

pyorient.exceptions.PyOrientWrongProtocolVersionException: Protocol version 37 is not supported yet by this client.

pyorient 1.5.5
orientdb 3.0.0

**Comment by alberto-exabel** [comment]
Same problem here.
pyorient 1.5.5
orientdb 3.0.2

**Comment by lscheinman77SAP** [comment]
Same issue.

**Comment by vinitshetty** [comment]
+1

**Comment by vinitshetty** [comment]
i am using 3.0.2 and pyorient 1.5.5

**Comment by dersmon** [comment]
Same issue using the current Docker image. 

I just wanted to try OrientDB and am not sure what a patch would encompass. No matter how complicated that may be, the Repository README states:
> Pyorient works with orientdb version 1.7 and later.

As a first measure: Could someone adjust that line, pointing to the last working OrientDB (2.x) version?

**Comment by Ostico** [comment]
@dersmon good point, since at moment i have not enough time to advance the version at least until august i think, this is the best option.

**Comment by softwareCobbler** [comment]
I changed `SUPPORTED_PROTOCOL` in `constants.py` from `36` to `37` and it connects...  now, is there going to be a bunch of buggy behavior?...

**Comment by Ostico** [comment]
Hi @softwareCobbler ,

not tried, not tested and I don't know which issues there could be. Could be useful to know which problems you find as a good starting point to short cut the driver improvement.

**Comment by Froskekongen** [comment]
@mogui: Is there any plan to update pyorient to support version 3.x reliably?

**Comment by wave-DmP** [comment]
@softwareCobbler  I tried your approach and got the following

> pyorient.exceptions.PyOrientDatabaseException: com.orientechnologies.orient.core.exception.OConfigurationException - You can use connect as first operation only for protocol  < 37 please use handshake for protocol >= 37

**Comment by TariqAHassan** [comment]
Hi @Ostico

Sorry to ping you directly, but any idea about the status of this issue?

**Comment by mogui** [comment]
HI Tariq,
We’ve been very busy with our everyday jobs, hopefully they will calm down soon, and we’re look into that soon, 
We’ve got some ideas on how to maintain compatibility, hang on :)

**Comment by montequie** [comment]
Hi Mogui,

Do you guys have any estimations about fixing it?
Or any proper python temporary solution to use mean while?

Thanks!!

**Comment by TariqAHassan** [comment]
@mogui 

I ended up deleted my comment because I found that commenting out the error message (or invoking `warning.warn()` instead raising in my case) will work. 

@klkludde @montequie: What I suggested above is not particularly elegant, but it should work as a stopgap measure.

**Comment by montequie** [comment]
@TariqAHassan 

Can you repost your suggestion? can't find it here

**Comment by montequie** [comment]
@TariqAHassan Thanks a lot!
It worked.

**Comment by dtanase** [comment]
+1
Any updates?

**Comment by dustyny** [comment]
+1

Let me add my encouragement to update to 3.x as well.. :)  

I'm a new user to OrientDB. I have a project that I'm working on and I'm nervous to build and then have to migrate to 3.x later on.

**Comment by shashwatwork** [comment]
Is there any fix for pyorient with Orientdb 3.0
I got error like **Protocol version 37 is not supported yet by this client**

**Comment by mn3mos** [comment]
+1

**Comment by upx86** [comment]
Is there any plan to fix this?

**Comment by laurendelong21** [comment]
+1

**Comment by mpomet** [comment]
If someone needs to use odb3 with python, just use master branch instead of the last release.
`pip install git+https://github.com/orientechnologies/pyorient`    

If you just start a new project, I recommend using a more up-to-date dbsm (redisgraph or other).

**Comment by CafeLungo** [comment]
I'm sorry, but... what the hell is going on here?

Try out orientdb -> use pyorient -> latest, and right off the bat I run into a non-starter because of a constant that hasn't been updated in almost 2 years now? (Along with other changes that haven't been implemented along with that constant bump)

**Comment by Schwarzfisch92** [comment]
it seems that this problem is back with the new update Orientdb 3.0.30
(I downloaded the latest version 3.0.30 with latest available Pyorient 1.5.5 ) but then it worked when i commented out:
 if self.protocol > SUPPORTED_PROTOCOL:
                raise PyOrientWrongProtocolVersionException(
                    "Protocol version " + str(self.protocol) +
                    " is not supported yet by this client.", [])

**Comment by jbonfardeci** [comment]
You can fix this by overriding the connect method in OrientSocket:
```
from pyorient import OrientSocket

class PySocket(OrientSocket):
    """
    Patch for OrientSocket
    John Bonfardeci
    2020-03-26
    """
    def __init__(self, host, port, protocol=36, timeout=300):
        """
        Override OrientSocket protocol version number
        to fix PyOrientWrongProtocolVersionException where:
            'Protocol version 37 is not supported yet by this client.'
        """
        super(PySocket, self).__init__(host, port)
        self.protocol = protocol
        self.host = host
        self.port = port
        self.timeout = timeout

    def connect(self):
        """
        Override OrientSocket connect method
        to fix PyOrientWrongProtocolVersionException where:
            'Protocol version 37 is not supported yet by this client.'
        """

        self._socket.settimeout(self.timeout)
        self._socket.connect( (self.host, self.port) )
        _value = self._socket.recv(2)

        if len(_value) != 2:
            self._socket.close()

        self.connected = True
```
then
```
HOST = "localhost"
PORT = 2424
DATABASE_NAME = "Covid19"
DB_USER = "admin"
DB_PWD = "admin"

socket = PySocket(HOST, PORT)
socket.connect()
client = OrientDB(socket)
client.db_open(DATABASE_NAME, DB_USER, DB_PWD)
```

**Comment by styk-tv** [comment]
I love fixes like these. OrientTechnologies just removed the if statement causing the error :)
https://github.com/orientechnologies/pyorient/commit/5bbe000acb0e65d49a3f85bd0ecf9d40ae155467

On the Orient 3+ document page above repo is linked as the repo to be used (just passing on the message) Probably because this bug is not fixed for the past 2 years.

Tested on:
Orientdb 3.0.30
Pyorient 1.5.5 

What you need to do is put direct request for component from the repo and not from the package manager.

In your requirements.txt instead of saying
`pyorient=-1.5.5.0` 
say
```
-e git+https://github.com/orientechnologies/pyorient.git@master#egg=pyorient
```

Now the connection as plain as below is working without the error. And be sure to check the first link with the link to have a laugh.
```
import pyorient
from pyorient.ogm import Graph, Config

graph = Graph(Config.from_url('plocal://127.0.0.1:2424/mydb', 'root', 'supersecret'))
```

**Comment by aniketgambhire** [comment]
Hi @rrmerugu. Install pyorient from source as given here [PyOrient Installation](https://orientdb.com/docs/last/PyOrient.html) after this you no longer will face this issue.

**Comment by sanvir10** [comment]
Or use **pyorient=1.5.1** with this version this work fine.

**Comment by imanabu** [comment]
@styk-tv's solution fixes the connection problem (thank you for that), but still there are several more issues that's the driver itself. 

## You need to now get a token and then set it, I did the following and got around that, 

        token = client.get_session_token()
        client.set_session_token(token)

## But, after that you will still get another exception if you do a Query.

      pyorient.exceptions.PyOrientBadMethodCallException: Unable to find command 'DbQueryMessage'
      
Granted these are different bugs, I need to warn that you will run into more issues even if you get around this specific issue on this bug.

**Comment by DatoAkobiaQualcommAtheros** [comment]
Getting same issue in 2024. No solutions, huh?
